src/api/app/services/workflows/artifacts_collector.rb

```diff
@@ -0,0 +1,40 @@
+module Workflows
+  class ArtifactsCollector
+    def initialize(step:, workflow_run_id:)
+      @step = step
+      @workflow_run_id = workflow_run_id
+    end
+
+    # TODO: test multibuilds and new approach with paths for configure_repositories
+    def call
+      artifacts = case @step.class.name
+                  when 'Workflow::Step::BranchPackageStep', 'Workflow::Step::LinkPackageStep'
+                    {
+                      source_project: @step.step_instructions[:source_project],
+                      source_package: @step.step_instructions[:source_package],
+                      target_project: @step.target_project_name,
+                      target_package: @step.target_package_name
+                    }
+                  when 'Workflow::Step::RebuildPackage'
+                    {
```

Comment:
This could also be written as `@step.step_instructions.slice(:project, :package)`.

Reply:
good catch!